src/pointer.rs

```diff
@@ -99,32 +42,50 @@ const fn is_valid_ptr(value: &str) -> bool {
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Pointer(str);
 
+impl Default for &'static Pointer {
+    fn default() -> Self {
+        Pointer::root()
+    }
+}
 impl core::fmt::Display for Pointer {
     fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         self.0.fmt(f)
     }
 }
-
 impl Pointer {
     /// Private constructor for strings that are known to be correctly encoded
     ///
     /// This is a zero-copy constructor
     fn new<S: AsRef<str> + ?Sized>(s: &S) -> &Self {
```

Comment:
I wonder if this and the other constructors should be `must_use` too?

`must_parse` probably can be left out, as I can imagine using it as a validator being useful in tests.

Reply:
Ah, I forgot about this one.

So after reading the lint description again, I'm actually okay with turning it off and rolling back all of the `#[must_use]` tags. They were added due to the pedantic [`clippy::must_use`)(https://rust-lang.github.io/rust-clippy/master/#/must_use) lint.